docs/source/federated-types/interfaces.mdx

```diff
@@ -0,0 +1,315 @@
+
+## Why is `@interfaceObject` necessary?
+
+Without the `@interfaceObject` directive and its associated composition logic, distributing an interface type's definition across subgraphs can impose continual maintenance requirements on your subgraph teams.
+
+Let's look at an example that _doesn't_ use `@interfaceObject`. Here, Subgraph A defines the `Media` interface, along with two implementing entities:
+
+```graphql title="Subgraph A"
+interface Media {
+  id: ID!
+  title: String!
+}
+
+type Book implements Media @key(fields: "id") {
+  id: ID!
+  title: String!
+  author: String!
+}
+
+type Movie implements Media @key(fields: "id") {
+  id: ID!
+  title: String!
+  director: String!
+}
+```
+
+Now, if Subgraph B wants to add a `reviews` field to the `Media` interface, it can't _just_ define that field:
+
+<p style="margin-bottom: 0">❌</p>
+
+```graphql title="Subgraph B"
+interface Media {
+  reviews: [Review!]!
+}
+
+type Review {
+  score: Int!
+}
+
+type Query {
+  bestRatedMedia: [Media!]!
+}
+```
+
+**This addition breaks composition.** In the supergraph schema, the `Media` interface now defines the `reviews` field, but neither `Book` nor `Movie` does!
+
+For this to work, Subgraph B _also_ needs to add the `reviews` field to _every entity_ that implements `Media`:
+
+<p style="margin-bottom: 0">⚠️</p>
+
+```graphql
+interface Media {
+  reviews: [Review!]!
+}
+
+type Review {
+  score: Int!
+}
+
+type Book implements Media @key(fields: "id") {
+  id: ID!
+  reviews: [Review!]!
+}
+
+type Movie implements Media @key(fields: "id") {
+  id: ID!
+  reviews: [Review!]!
+}
+```
+
+This resolves our _current_ composition error, but composition will break _again_ whenever Subgraph A defines a _new_ entity that implements `Media`:
+
+```graphql title="Subgraph A"
+type Podcast implements Media @key(fields: "id") {
+  id: ID!
+  title: String!
+}
+```
+
+To prevent these composition errors, the teams maintaining Subgraph A and Subgraph B need to coordinate their schema changes _every time_ a new implementation of `Media` is created. Imagine how complex that coordination becomes if the definition of `Media` is instead distributed across _ten_ subgraphs!
+
+In summary, Subgraph B shouldn't need to know every possible _kind_ of `Media` that exists in your supergraph. Instead, it should generically know how to fetch reviews for _any_ kind of `Media`. _This_ is the relationship that entity interfaces and `@interfaceObject` provide, as demonstrated in the [example above](#example-schemas).
+
+
+### Are there alternatives to using `@interfaceObject`?
+
+The primary alternative to using `@interfaceObject` is to use the discouraged strategy described in the [previous section](#why-is-interfaceobject-necessary). This requires duplicating _all_ implementations of a given interface in each subgraph that contributes fields to that interface.
```

Comment:
I'm not sure where to put it, but the caveat to this strategy that that subgraph must be able to resolve the interface's type. If the upstream data store only contains the key fields, but doesn't know if something if a Book or Movie, this strategy isn't even possible.